leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/MyDevice.java

```diff
@@ -90,10 +91,10 @@ public ReadResponse read(ServerIdentity identity, int resourceid) {
     }
 
     @Override
-    public ExecuteResponse execute(ServerIdentity identity, int resourceid, String params) {
+    public ExecuteResponse execute(ServerIdentity identity, int resourceid, Arguments arguments) {
         String withParams = null;
-        if (params != null && params.length() != 0)
-            withParams = " with params " + params;
+        if (arguments.size() != 0)
+            withParams = " with params " + arguments;
```

Comment:
Do not hesitate to replace "parameters` by `arguments` for Execute Request as this seems to be the terminology used in the specification.